cmd/end2endtest/interruptible.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"errors"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	vapi "go.vocdoni.io/dvote/api"
+	"go.vocdoni.io/dvote/apiclient"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+	"go.vocdoni.io/dvote/log"
+)
+
+func init() {
+	ops["uninterruptiblelection"] = operation{
+		test:        &E2EUninterruptibleElection{},
+		description: "Publishes a non-interruptible election to validate that the election cannot be interrupted changing the status to ENDED or PAUSED or CANCELED or RESULTS",
+		example:     os.Args[0] + " --operation=uninterruptiblelection --votes=1000",
+	}
+}
+
+var _ VochainTest = (*E2EUninterruptibleElection)(nil)
+
+type E2EUninterruptibleElection struct {
+	e2eElection
+}
+
+func (t *E2EUninterruptibleElection) Setup(api *apiclient.HTTPclient, c *config) error {
+	t.api = api
+	t.config = c
+
+	ed := newTestElectionDescription()
+	ed.ElectionType = vapi.ElectionType{
+		Autostart:     true,
+		Interruptible: false,
+	}
+	ed.VoteType = vapi.VoteType{MaxVoteOverwrites: 1}
+	ed.Census = vapi.CensusTypeDescription{Type: vapi.CensusTypeWeighted}
+
+	if err := t.setupElection(ed); err != nil {
+		return err
+	}
+
+	log.Debugf("election details: %+v", *t.election)
+	return nil
+}
+
+func (t *E2EUninterruptibleElection) Teardown() error {
+	// nothing to do here
+	return nil
+}
+
+func (t *E2EUninterruptibleElection) Run() error {
+	c := t.config
+	api := t.api
+
+	// Send the votes (parallelized)
+	startTime := time.Now()
+	wg := sync.WaitGroup{}
+	voteAccounts := func(accounts []*ethereum.SignKeys, wg *sync.WaitGroup) {
+		defer wg.Done()
+		log.Infof("sending %d votes", len(accounts))
+
+		votesSent := 0
+		contextDeadlines := 0
+		for i, acc := range accounts {
+			ctxDeadline, err := t.sendVote(voteInfo{voterAccount: acc, choice: []int{i % 2}}, nil)
+			if err != nil {
+				log.Error(err)
+				break
+			}
+			contextDeadlines += ctxDeadline
+			votesSent++
+		}
+		log.Infof("successfully sent %d votes... got %d HTTP errors", votesSent, contextDeadlines)
+		time.Sleep(time.Second * 4)
+	}
+
+	pcount := c.nvotes / c.parallelCount
+	for i := 0; i < len(t.voterAccounts); i += pcount {
+		end := i + pcount
+		if end > len(t.voterAccounts) {
+			end = len(t.voterAccounts)
+		}
+		wg.Add(1)
+		go voteAccounts(t.voterAccounts[i:end], &wg)
+	}
+
+	wg.Wait()
+	log.Infof("%d votes submitted successfully, took %s (%d votes/second)",
+		c.nvotes, time.Since(startTime), int(float64(c.nvotes)/time.Since(startTime).Seconds()))
```

Comment:
i believe you can also skip the votes part. just set up the election (READY) and then simply try to interrupt it.